includes/class-wc-post-types.php

```diff
@@ -29,6 +29,7 @@ public static function init() {
 		add_action( 'init', array( __CLASS__, 'register_post_status' ), 9 );
 		add_action( 'init', array( __CLASS__, 'support_jetpack_omnisearch' ) );
 		add_filter( 'rest_api_allowed_post_types', array( __CLASS__, 'rest_api_allowed_post_types' ) );
+		add_action( 'woocommerce_flush_rewrite_rules', array( __CLASS__, 'flush_rewrite_rules' ) );
```

Comment:
yes, its right. but i think if you must do more on flush later you can add it to the method. Feel free to change it.